tidy3d/components/eme/sweep.py

```diff
@@ -81,4 +83,46 @@ def num_sweep(self) -> pd.PositiveInt:
         return len(self.freq_scale_factors)
 
 
-EMESweepSpecType = Union[EMELengthSweep, EMEModeSweep, EMEFreqSweep]
+class EMEPeriodicitySweep(EMESweepSpec):
+    """Spec for sweeping number of repetitions of EME subgrids.
+    Useful for simulating long periodic structures like Bragg gratings,
+    as it allows the EME solver to reuse the modes and cell interface
+    scattering matrices.
+
+    Compared to setting ``num_reps`` directly in the ``eme_grid_spec``,
+    this sweep spec allows varying the number of repetitions,
+    effectively simulating multiple structures in a single EME simulation.
+
+    Example
+    -------
+    >>> n_list = [1, 50, 100]
+    >>> sweep_spec = EMEPeriodicitySweep(num_reps=[{"unit_cell": n} for n in n_list])
```

Comment:
I'm not sure how I feel about using the `name` attribute for functional purposes. But this is not the first place in the code this happens. Just curious - is it used anywhere else in EME, and did you consider any alternatives?

Reply:
It’s only here. I considered forcing the sweep num_reps to be a list of tuples, each of which contained a num_reps for the current grid spec, as well as a list of num_reps for the sub grids if it is composite (as a recursive data structure in the same format as the grid spec). But this was way too cumbersome